ironfish-cli/src/utils/chainport/utils.ts

```diff
@@ -0,0 +1,111 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { RpcWalletTransaction, TransactionType } from '@ironfish/sdk'
+import { getConfig } from './config'
+import { ChainportMemoMetadata } from './metadata'
+
+type ChainportTransactionDetails = {
+  chainportNetworkId: number
+  address: string
+}
+
+type NotChainportTransaction = {
+  isChainportTransaction: false
+}
+
+type ChainportTransaction = {
+  isChainportTransaction: true
+  type: TransactionType.SEND | TransactionType.RECEIVE
+  details: ChainportTransactionDetails
+}
+
+type ChainportTransactionResult = ChainportTransaction | NotChainportTransaction
+
+export const isChainportTransaction = (
+  networkId: number,
+  transaction: RpcWalletTransaction,
+): ChainportTransactionResult => {
+  const config = getConfig(networkId)
+
+  if (transaction.type === TransactionType.SEND) {
+    return isOutgoingChainportBridgeTransaction(transaction, config)
+  }
+  if (transaction.type === TransactionType.RECEIVE) {
+    return isIncomingChainportBridgeTransaction(transaction, config)
+  }
+  return {
+    isChainportTransaction: false,
+  }
+}
+
+const isIncomingChainportBridgeTransaction = (
+  transaction: RpcWalletTransaction,
+  config: { incomingAddresses: string[] },
+): ChainportTransactionResult => {
+  const bridgeNote = transaction.notes?.[0]
+
+  if (!bridgeNote || !isAddressInList(bridgeNote.sender, config.incomingAddresses)) {
+    return {
+      isChainportTransaction: false,
+    }
+  }
+
+  const [sourceNetwork, address, _] = ChainportMemoMetadata.decode(
+    Buffer.from(bridgeNote.memoHex).toString(),
+  )
+
+  return {
+    isChainportTransaction: true,
+    type: TransactionType.RECEIVE,
+    details: {
+      chainportNetworkId: sourceNetwork,
+      address: address,
+    },
+  }
+}
+
+const isOutgoingChainportBridgeTransaction = (
+  transaction: RpcWalletTransaction,
+  config: { outgoingAddresses: string[] },
+): ChainportTransactionResult => {
+  if (!transaction.notes || transaction.notes.length < 2) {
+    return {
+      isChainportTransaction: false,
+    }
+  }
+
+  if (!transaction.notes.find((note) => note.memo === '{"type": "fee_payment"}')) {
+    return {
+      isChainportTransaction: false,
+    }
+  }
+
+  const bridgeNote = transaction.notes.find((note) =>
+    isAddressInList(note.owner, config.outgoingAddresses),
+  )
+
+  if (!bridgeNote) {
+    return {
+      isChainportTransaction: false,
+    }
+  }
+
+  const [sourceNetwork, address, _] = ChainportMemoMetadata.decode(
+    Buffer.from(bridgeNote.memoHex).toString(),
```

Comment:
why do we need to convert string -> Buffer -> string here?

Reply:
fixed